### DISCUSSION TITLE
Is hugo-theme-stack supports branch bundles ?

### DISCUSSION THREAD

**Discussion by nodauf** [CaiJimmy/hugo-theme-stack#623] [Q&A] [answered]
In the issue #570 you stated that the issue with the images was not a real problem and you give a quote about the **leaf bundles**.

Thus, I wonder if the [branches bundles](https://gohugo.io/content-management/page-bundles/#branch-bundles) are supported by this theme.
```
content/
├── branch-bundle-1
│   ├── branch-content1.md
│   ├── branch-content2.md
│   ├── image1.jpg
│   ├── image2.png
│   └── _index.md
└── branch-bundle-2
    ├── _index.md
    └── a-leaf-bundle
        └── index.md
```

**Comment by CaiJimmy** [comment]
> Thus, I wonder if the [branches bundles](https://gohugo.io/content-management/page-bundles/#branch-bundles) are supported by this theme.

Should work with https://github.com/CaiJimmy/hugo-theme-stack/pull/81.

**Comment by nodauf** [comment]
No, I can't make it work. I tried with `image` attribute for the cover picture and also images inside a post.

My folder is structure is like this:
```
post/
├── branch1
│   ├── article1.md
│   ├── article2md
│   ├── image1.png
│   └── _index.md
└── ...
```

Inside article1.md, I insert my image:
```
![My image](image1.png)
```

When I am trying to load the post, the image can't be loaded. Its path is http://localhost:1313/image1.png

My tests on the cover image shows that if I am renaming the index.md file, the image cannot be loaded anymore.

**Comment by CaiJimmy** [comment]
In order to use page resources (images), the markdown file has to be named `index.md`. The Hugo documentation you mentioned initially has specified that clearly:

```
content/
├── about
│   ├── index.md
├── posts
│   ├── my-post
│   │   ├── content1.md
│   │   ├── content2.md
│   │   ├── image1.jpg
│   │   ├── image2.png
│   │   └── index.md
│   └── my-other-post
│       └── index.md
│
└── another-section
    ├── ..
    └── not-a-leaf-bundle
        ├── ..
        └── another-leaf-bundle
            └── index.md
```

> image1
> This image is a page resource of my-post and **only** available in my-post/index.md resources.

https://gohugo.io/content-management/page-bundles/#leaf-bundles

**Comment by CaiJimmy** [comment]
You can have many nested branch bundles as you like. But in order to access a page resource (image), the file has to be called `index.md` (a leaf bundle). That Hugo's restriction, not theme's.

**Comment by nodauf** [comment]
I first came here because I migrate from [hugo-clarity](https://github.com/chipzoller/hugo-clarity) where everything was working fine (I was able to include an image in others file than index.md).

**Comment by CaiJimmy** [comment, marked as answer]
I just checked the code of that theme, and it uses some tricks to read the image outside the leaf bundle, not only `.Page.Resources.GetMatch` which is what I use in this theme.

https://github.com/chipzoller/hugo-clarity/blob/e68a523254487da99652f94f99e23eb781f95330/layouts/_default/_markup/render-image.html#L35-L44

I prefer to keep the thing as it is right now, just not to mess it up...

**Comment by nodauf** [comment]
Ok thank you for your time. Didn't know they use tricks.
